poi-ooxml/src/main/java/org/apache/poi/openxml4j/util/ZipSecureFile.java

```diff
@@ -129,13 +135,22 @@ public static long getMaxTextSize() {
     }
 
     public ZipSecureFile(File file) throws IOException {
-        super(file);
-        this.fileName = file.getAbsolutePath();
+        this(file.toPath());
+    }
+
+    public ZipSecureFile(Path path) throws IOException {
+        super(
+            Files.newByteChannel(path, EnumSet.of(StandardOpenOption.READ)),
+            path.toAbsolutePath().toString(),
+            StandardCharsets.UTF_8.name(),
+            true,
+            false
+        );
+        this.fileName = path.toAbsolutePath().toString();
     }
 
     public ZipSecureFile(String name) throws IOException {
-        super(name);
-        this.fileName = new File(name).getAbsolutePath();
+        this(new File(name));
```

Comment:
I like the clarity of this and ordinarily support constructor chaining, but it seems a bit strange to convert to a file when we could just as easily say 

```suggestion
        this(Paths.get(name));
```

Reply:
i agree with the change - i didnt accept this suggestion because it also needs the import :P